elastic/kibana
# [Fleet] Tighten policy permissions, take II

## Summary

Closes #64634.

This PR refines the work done in #94591. It narrows down the API key permissions that each agent needs based on the integrations that the policy uses.

To do so we look into the policy's `PacakgePolicy` and the enabled inputs. We link that with the package `DataStream` to determine which indices it uses and which permissions it needs. The permissions will be specified in the package's `data_streams/*/manifest.yml` (pending proposal).

At the moment of writing no packages specify what permissions they need, so I'm using the permissions that are necessary right now as defaults.

### Testing

To test this PR:

1. Go to Fleet > Policies
2. Click on any policy
3. Click on Actions > View policy

The policy should have an `output_permissions` block, with several permissions per package instead of the generic `logs-*`, `metrics-*`, etc.


### Checklist

Delete any items that are not applicable to this PR.
